app/server/utils/rate_limit.py

```diff
@@ -0,0 +1,29 @@
+from server import red
+from flask import current_app
+
+def rate_limit(key, rate):
+    """Function to impose ratelimits on an API. Pass it a key, and it'll only allow that key `rate` times per hour
+    key: Key on which to rate limit (like a username)
+    rate: Number of allowed requests per hour
+    
+    Returns True if rate limited
+    """
+
+    # Don't ratelimit our unit tests!
+    if current_app.config['IS_TEST']:
+        return False
+
+    # Check if the key has been tried recently
+    if red.exists(key):
+        attempts = int(red.get(key))
+        if attempts > rate:
+            # If rate limited, return how long (minutes) until you can try again
+            ttl = int(red.ttl(key)/60)
+            return ttl
+        red.set(key, attempts+1, keepttl=True)
```

Comment:
This line is causing issues for me. The first API request works, but them I'm getting the following traceback and error:

```
Traceback (most recent call last):
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 90, in sentry_patched_wsgi_app
    environ, start_response
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 129, in __call__
    reraise(*_capture_exception(hub))
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/_compat.py", line 55, in reraise
    raise value
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/integrations/wsgi.py", line 125, in __call__
    _sentry_start_response, start_response, transaction
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/integrations/flask.py", line 89, in <lambda>
    return SentryWsgiMiddleware(lambda *a, **kw: old_app(self, *a, **kw))(
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 2464, in __call__
    return self.wsgi_app(environ, start_response)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 2450, in wsgi_app
    response = self.handle_exception(e)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 1867, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask_cors/extension.py", line 165, in wrapped_function
    return cors_after_request(app.make_response(f(*args, **kwargs)))
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/views.py", line 89, in view
    return self.dispatch_request(*args, **kwargs)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/flask/views.py", line 163, in dispatch_request
    return meth(*args, **kwargs)
  File "/Users/tristancole/PycharmProjects/SempoBlockchain/app/server/api/auth_api.py", line 334, in post
    limit = rate_limit("login_"+ratelimit_key, 25)
  File "/Users/tristancole/PycharmProjects/SempoBlockchain/app/server/utils/rate_limit.py", line 23, in rate_limit
    red.set(key, attempts+1, keepttl=True)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/redis/client.py", line 1801, in set
    return self.execute_command('SET', *pieces)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/sentry_sdk/integrations/redis.py", line 101, in sentry_patched_execute_command
    return old_execute_command(self, name, *args, **kwargs)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/redis/client.py", line 901, in execute_command
    return self.parse_response(conn, command_name, **options)
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/redis/client.py", line 915, in parse_response
    response = connection.read_response()
  File "/Users/tristancole/VirtualEnvs/SempoBlockchain_env10/lib/python3.7/site-packages/redis/connection.py", line 756, in read_response
    raise response
redis.exceptions.ResponseError: syntax error
```
https://drive.google.com/file/d/1LYwencscvSACFjkYWDLlyckp44wDHNdP/view?usp=sharing